In [1]:
import numpy as np
import pandas as pd

In [4]:
df = pd.read_csv('sales_data_with_discounts.csv')

In [20]:
df.sample(3)

,Date,Day,SKU,City,Volume,BU,Brand,Model,Avg Price,Total Sales Value,Discount Rate (%),Discount Amount,Net Sales Value
274,10-04-2021,Saturday,M05,C,4,Mobiles,YouM,YM-98,8100,32400,16.553848,5363.446806,27036.553194
419,14-04-2021,Wednesday,L10,C,2,Lifestyle,Jeera,M-Shoes,3100,6200,17.414974,1079.728376,5120.271624
140,05-04-2021,Monday,L01,C,14,Lifestyle,Jeera,M- T Shirts,450,6300,17.194704,1083.266375,5216.733625


In [6]:
df.dtypes

Date                  object
Day                   object
SKU                   object
City                  object
Volume                 int64
BU                    object
Brand                 object
Model                 object
Avg Price              int64
Total Sales Value      int64
Discount Rate (%)    float64
Discount Amount      float64
Net Sales Value      float64
dtype: object

In [7]:
df.mean(numeric_only = True)

Volume                   5.066667
Avg Price            10453.433333
Total Sales Value    33812.835556
Discount Rate (%)       15.155242
Discount Amount       3346.499424
Net Sales Value      30466.336131
dtype: float64

#### on average, approximately 5 units of product units were sold per transaction.
#### the average selling price of the product accross all transactions is 10453.
#### Before applying Discount the average value of a sales transaction was around 33812.
#### customers recieve an average discount of 15 % on the total sales value.
#### On average the amount 3346 was deducted from each transaction as a discount.
#### After discount the average amount paid by the customers per transaction is 30466.

In [8]:
df.median(numeric_only = True)

Volume                  4.000000
Avg Price            1450.000000
Total Sales Value    5700.000000
Discount Rate (%)      16.577766
Discount Amount       988.933733
Net Sales Value      4677.788059
dtype: float64

#### Half of the sales transactions involved 4 or fewer units, while the other half involved more than 4 units.
#### The median selling price per unit was ₹1,450, meaning 50% of the products were priced below this value and 50% above it.
#### Half of the transactions had a total sales value of ₹5,700 or less, and the other half exceeded this amount.
#### The typical discount offered was approximately 16.58%, with half the transactions receiving a lower discount and half receiving a higher discount.
#### In a typical transaction, the discount amount was approximately ₹989.
#### The median amount paid by customers after discounts was approximately ₹4,678 per transaction.

In [16]:
df.mode(numeric_only = True, dropna = True)

,Volume,Avg Price,Total Sales Value,Discount Rate (%),Discount Amount,Net Sales Value
0,3.0,400.0,24300.0,5.007822,69.177942,326.974801
1,NaN,450.0,NaN,5.055218,73.025199,330.822058
2,NaN,500.0,NaN,5.059801,93.649161,466.350839
3,NaN,1300.0,NaN,5.072124,94.682735,485.317265
4,NaN,8100.0,NaN,5.084108,102.705757,496.607104
...,...,...,...,...,...,...
445,NaN,NaN,NaN,19.940610,17900.983733,162603.186379
446,NaN,NaN,NaN,19.965340,21153.498820,163915.971000
447,NaN,NaN,NaN,19.976879,21496.675367,163967.080070
448,NaN,NaN,NaN,19.982257,25328.224204,166263.161348


In [17]:
df.var(numeric_only = True)

Volume               1.790646e+01
Avg Price            3.268830e+08
Total Sales Value    2.553794e+09
Discount Rate (%)    1.781348e+01
Discount Amount      2.033922e+07
Net Sales Value      2.149125e+09
dtype: float64

#### The number of units sold per transaction shows moderate variation, indicating that sales quantities differ across transactions.
#### The selling prices vary greatly, indicating the presence of products with a wide range of prices, from low-cost to high-cost items.
#### There is a very high variation in total sales values, suggesting that transaction amounts differ significantly, likely due to differences in product prices and quantities sold.
#### The discount percentages show relatively low variation, indicating that discounts offered are fairly consistent across transactions.
#### The discount amounts vary considerably because higher-value transactions receive larger discounts.
#### The final sales values after discounts also exhibit high variability, reflecting substantial differences in transaction values.

In [18]:
df.std(numeric_only = True)

Volume                   4.231602
Avg Price            18079.904840
Total Sales Value    50535.074173
Discount Rate (%)        4.220602
Discount Amount       4509.902963
Net Sales Value      46358.656624
dtype: float64